### PR TITLE
fix ts-node dev server

### PR DESCRIPTION
**Pull request by darasus**
Fix ts-node dev server

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -92,7 +92,7 @@
     "workerpool": "^6.2.0"
   },
   "scripts": {
-    "dev": "DEBUG='bp:*' node --preserve-symlinks --trace-deprecation index",
+    "dev": "DEBUG='bp:*' nodemon",
```
Comment (igordanchenko) [inline package.json:95]: The "start" script also uses `node index` which no longer works.

File: tsconfig.server.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "module": "commonjs",
+    "outDir": "dist",
```
Comment (igordanchenko) [inline tsconfig.server.json:5]: Do we really need "outDir", "noEmit", "isolatedModules", and "include" in this config?

Can't the config be simplified to 

```
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "module": "commonjs",
    "target": "es2017"
  }
}
```
?

**Comment by pastelsky** [comment]
Opinions on replacing `ts-node`, `nodemon` etc with https://github.com/esbuild-kit/tsx ?

**Comment by pastelsky** [comment]
We should probably add a ping test for this as well in our suite so we don't regress on server start in the future.

**Comment by darasus** [comment]
@pastelsky good suggestion, I've updated PR to use `npx esno`, it's a thing wrapper around tsx

**Review by pastelsky — APPROVED** [review]

**Comment by pastelsky** [comment]
I just merged this and realized that `esno` (or any ESBuild related solutions) don't work with the canvas library.
@darasus how were you able to run this? 

I'm switching back to `ts-node` for the time being as our mainline is a broken state due to this issue, plus some errors from the `function getEnv` refactor.

**Comment by igordanchenko** [comment]
I'm afraid the main branch is still broken. 

2 issues with this PR:

1) It's probably not a good idea to use 'npx' in any scripts (especially in `yarn prod`) as it may block awaiting user input :

```
% yarn prod

<build step>

Need to install the following packages:
  esno
Ok to proceed? (y) 
```

2) The server doesn't start:

```
% yarn dev
{
  detail: undefined,
  id: 'suspicious-boolean-not',
  location: {
    column: 8,
    file: '<project_root>/bundlephobia/server/middlewares/results/error.middleware.js',
    length: 7,
    line: 39,
    lineText: "    if (!'name' in err) {",
    namespace: '',
    suggestion: "(!'name')"
  },
  notes: [
    {
      location: null,
      text: 'The code "!x in y" is parsed as "(!x) in y". You need to insert parentheses to get "!(x in y)" instead.'
    }
  ],
  pluginName: '',
  text: 'Suspicious use of the "!" operator inside the "in" operator'
}
<project_root>/bundlephobia/node_modules/fabric/dist/fabric.js:37
  DOMParser = fabric.window.DOMParser;
  ^

ReferenceError: DOMParser is not defined
    at fabric (/Users/igor/Development/github/bundlephobia/node_modules/fabric/dist/fabric.js:37:3)
    at Object.<anonymous> (/Users/igor/Development/github/bundlephobia/node_modules/fabric/dist/fabric.js:31086:4)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.v (/Users/igor/.npm/_npx/1054ab95ad6fbcfc/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:901)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at null.<anonymous> (/Users/igor/Development/github/bundlephobia/utils/draw.utils.js:2:20)
    at Object.<anonymous> (/Users/igor/Development/github/bundlephobia/utils/draw.utils.js:227:33)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.v (/Users/igor/.npm/_npx/1054ab95ad6fbcfc/node_modules/@esbuild-kit/cjs-loader/dist/index.js:1:901)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
```

**Comment by darasus** [comment]
@pastelsky my bad, shall I prep `ts-node` PR?

**Comment by darasus** [comment]
Only tested home page on `esno`.

**Comment by pastelsky** [comment]
> Only tested home page on esno.
Not sure how though — since the devloop seems to be broken for me completely. 

I already have a PR ready for a fix — should be done in a few mins.

**Comment by darasus** [comment]
🤷
CleanShot 2022-11-11 at 16 21 41@2x

**Comment by pastelsky** [comment]
https://github.com/pastelsky/bundlephobia/pull/722

**Pull request merged**